crates/cluster/src/ring.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2025 ScopeDB <contact@scopedb.io>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::collections::BTreeSet;
+
+use sha2::Digest;
+use sha2::Sha256;
+use sha2::digest::OutputSizeUser;
+use sha2::digest::generic_array::GenericArray;
+
+const DEFAULT_REPLICA_COUNT: usize = 100;
+
+type Sha256Digest = GenericArray<u8, <Sha256 as OutputSizeUser>::OutputSize>;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Ring<T> {
```

Comment:
```suggestion
#[derive(Debug, Clone, PartialEq, Eq, Hash)]
pub struct MembershipRing<T> {
```

TODO: document.

`Ring` can be easily overridden.

Reply:
This is idiom name, I think it's fine. But let's make it more clear.

Reply:
Make sense.